### PR TITLE
Make Jsonl Corpus reader path optional again

### PR DESCRIPTION
**Pull request by polm**
## Description

In #8396 the path arg to the JsonlCorpus was made optional. This was done so that you could use a config with a pretraining section for training without including the raw text path. However, #9167 made the path non-optional again, apparently inadvertently. 

### Types of change

minor fix

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/training/corpus.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def create_docbin_reader(
 
 @util.registry.readers("spacy.JsonlCorpus.v1")
 def create_jsonl_reader(
-    path: Union[str, Path], min_length: int = 0, max_length: int = 0, limit: int = 0
+    path: Union[None, str, Path], min_length: int = 0, max_length: int = 0, limit: int = 0
```
Comment (svlandeg) [inline spacy/training/corpus.py:44]:
```suggestion
    path: Optional[Union[str, Path]], min_length: int = 0, max_length: int = 0, limit: int = 0
```

**Comment by svlandeg** [comment]
Ah, this was probably an oversight on the mypy PR, yes. To make the tests succeed, you'll have to make sure that the `path` argument of the `JsonlCorpus` constructor matches this change though. And let's use `Optional` instead of `Union[None,...]` as we do in other places of the code base.

**Comment by svlandeg** [comment]
Oh I think you pushed the edit to the wrong PR 🙈 https://github.com/explosion/spaCy/pull/9639

**Comment by polm** [comment]
Gah, thanks for catching that! Should be fixed now.

**Review by svlandeg — APPROVED** [review]

**Pull request merged**